In [1]:
import math
from scipy import integrate
from IPython.display import Math

# Week 1

# Exercises

## Part 1: Circuit Elements

### S1E1.5: Simple Power

In the circuit shown below, the strength of the source is Vs=10V, and the resistance of the resistor is R=50ohms.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/58c80a1750b4ea92c3d83507f3d8ae9a/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_simple-circuit.gif)

#### Question A

What is the power dissipated in the resistor (in Watts)?

#### Solution

Using the relation P = V^2 / R:

In [2]:
V = 10
R = 50
PR = V**2 / R
print(str(PR), 'watts')

2.0 watts


#### Question B

What is the power entering the source (in Watts)?

#### Solution

The power entering the source is the negative of the power dissapated across the resistor

In [3]:
PS = -PR
print(str(PS), 'watts')

-2.0 watts


### S1E2: Power

The picture shows a resistor connected to some unknown network N. The resistor is immersed in an isolated water bath, and its temperature is observed and recorded. The resistor has resistance R=8.0ohms. By observing the rate of increase of the temperature in the water bath, it is determined that the power dissipated in the resistor is 11.0 W.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/88cf1cef3b66706c63d46b99d8436c4b/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_resistor-load.gif)

#### Question A

Assuming that the voltage across the resistor is constant, what is the voltage  (in Volts) across the resistor?

#### Solution

Use the relation P = V^2/R. Rearranging, V=sqrt(P*R)

In [4]:
P = 11.0
R = 8.0
V = math.sqrt(P * R)
print(str(V), 'volts')

9.38083151964686 volts


#### Question B

What is the current I (in Amperes) entering the network N from the resistor?

#### Solution

We have P and V. Use the relation I=P/V. Because current flows away from the resistor, the result is negative.

In [5]:
I = -P/V
print(str(I), 'amps')

-1.1726039399558574 amps


### S1E3: AC Power

The plot shows 1/10 second of the voltage waveform of a 120V 60Hz AC (Alternating Current) power circuit, like that delivered to residences in the United States.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/f0350b44586727b7d5ec3e01f5580b81/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_120V60Hz.gif)

The actual voltage is 120 * sqrt(2) * cos(2*pi*60*t) Volts. If we apply this voltage across a resistor of resistance 110.0ohms the resistor will dissipate a time-varying power. 

#### Question A

What is the peak power (in Watts) dissipated by the resistor?

#### Solution

For peak power, just calculate the voltage, then use P = V^2/R

In [6]:
R = 110.0
V = 120 * math.sqrt(2) * math.cos(2 * math.pi * 60)
print('V:', str(V), 'volts')
P = V**2/R
print('P:', str(P), 'watts')

V: 169.7056274847714 volts
P: 261.81818181818187 watts


#### Question B

What is the average power (in Watts) dissipated by the resistor? (Hint: you compute the average power by integrating the instantaneous power over one cycle of the waveform and dividing the result by the length of the cycle.)

#### Solution

In [7]:
Math(r'P_{avg} = \frac{V^2}{R}= \frac{\int_{0}^{1}(120 \sqrt{2} \cos(2 \pi 60 t))^2 dt}{110} = \frac{14400}{110} = 130.91')

<IPython.core.display.Math object>

In [8]:
def f(t):
    return (120 * math.sqrt(2) * math.cos(2*math.pi*60*t))**2
    
Vsquared, err = integrate.quad(f,0,1)
P = Vsquared/R
print(P, 'watts')

130.90909090909088 watts


#### Question C

What would be the power (in Watts) dissipated by the resistor if the voltage was a constant value of 120V?

#### Solution

Simple P=V^2/R

In [9]:
V = 120
P = V**2 / R
print(P, 'watts')

130.9090909090909 watts


### S1E5: KVL-0

Joe was debugging part of an experimental apparatus, probing around with his voltmeter. Part of the apparatus had two obvious resistors in series with an unknown element, as shown in the diagram below:

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/dcda4e5e49eb702293a888804d2d4b6d/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_KVL-0.gif)

The unknown element is hard to reach, so Joe put the negative (black) probe of his voltmeter at the interconnection of the two obvious resistors and then put the positive (red) probe at the other end of each resistor, measuring V1=1.4V and V2=0.9V.

#### Question

What is the voltage (in Volts) V3 measured across the unknown element?

#### Solution

Just subtract the two Voltages

In [10]:
V1 = 1.4
V2 = 0.9
V3 = V1 - V2
print(V3, 'volts')

0.4999999999999999 volts


### S1E6: KVL

In the circuit shown there are four unknown elements and an independent voltage source.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/cab77aa01b521b3931e74d5447e20b36/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_kvl.gif)

The strength of the source V is given. Also, there are two known branch voltages: we know V1 and V2.

#### Question A

In terms of the known voltages, write an algebraic expression for the branch voltage V3. Be careful, algebraic expressions are case sensitive.

#### Solution

We know V, V1, and V2. So V3 equals (in the proper syntax):

V - v1

#### Question B

Write an algebraic expression for the branch voltage V4:

#### Solution

v2 - V

#### Question C

Write an algebraic expression for the branch voltage V5:

#### Solution

v1 + v2 - V

### S1E7: KCL-0

While further poking around in his experiment, Joe found an exposed interconnect. It connected the positive output of the 5V power supply, a 3.3ohm resistor, and two unknown 2-terminal elements, as shown:

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/56af05602863830c8179890b29be1a64/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_KCL-0.gif)

He couldn't trace where the other ends of the elements went. However, he had a nifty (very expensive!) Hall-effect clamp-on ammeter that he used to measure the three of the four currents entering the elements. He didn't have enough space to measure the current entering the fourth element. He found that:

I1 = -0.7A

I3 = 3.0A

I4 = 1.3A

#### Question A

What was the current (in Amperes) I2 into the fourth element?

#### Solution

In [11]:
I1 = -0.7
I3 = 3.0
I4 = 1.3

# First solve for I5. Because the node at I5,I3,I4 must be zero, 
# and I5 flows into the node while I4 and I3 flow out, we can just add 
# I4 and I3
I5 = I4 + I3
print('I5:', I5, 'amps')

# Now that we have I5, solve for I2. The node at I1, I2, I5 must all be
# zero. Because all three flow away from the node, we add I1 and I5,
# then negate.
I2 = -(I1 + I5)
print('I2:', I2, 'amps')

I5: 4.3 amps
I2: -3.5999999999999996 amps


#### Question B

Joe managed to get his ammeter probe around a wire and measured . What current (in Amperes) did he measure?

#### Solution

We already figured out that I5 = 4.3A above

### S1E8: KCL

In the circuit shown there are six unknown elements. By measurements we have values for three of the branch currents: I1, I2, and I3:

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/7e9c1e319be42b11fb6c5ca5aad7d2b2/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_kcl.gif)

#### Question A: 

In terms of the known currents, write an algebraic expression for the current I4

#### Solution

i2 - i3

#### Question B

Write an algebraic expression for the current

#### Solution

We already know that i4 = i2 - i3

Looking at the note I1, I4, I5, we have the equation:

i5 = -(i1 + i4)

Subbing in i4, we get

-(i1+i2-i3)

#### Question C

Write an algebraic expression for the current I6

#### Solution

At the node, we have:

i6 = i5 - i3

We sub in the value for i5:

-(i1+i2-i3) - i3

Simplifying:

-i1 -i2 +i3 -i3

Finally:

-i1 -i2

### S1E9: Battery Model

A real battery is not an ideal independent voltage source. A voltage source is an appropriate idealization of the battery if the load on the battery is negligible. A better model for a battery is a voltage source in series with an ideal linear resistor whose resistance varies with temperature. Sometimes even better models are required - this [Energizer technical bulletin](http://data.energizer.com/PDFs/BatteryIR.pdf) gives more information. However, let's use the simple model of a linear resistor in series with an ideal independent voltage source, as in the figure.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/28917c942dfcf7a48315d5c754797cc1/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_battery-model.gif)

It is suggested in section 1.5.1 of the textbook that to increase the current-capacity of a battery without increasing the voltage at the terminals we can connect batteries of the same voltage in parallel. Let's examine this using our model.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/41cf8c3f78c807faa2ff5cf15bdfbd5b/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_two-batteries.gif)

#### Question A

Let's assume that both component batteries have the same voltage V1 = V2 = 1.5. The internal resistances of small batteries are about 0.2ohms, but they vary a bit. Let's assume that R1=0.25ohms  and R2=0.32ohms. What is the open-circuit voltage (in Volts) V of the combination?

#### Solution

This is just 1.5V. The voltage will not drop unless there is current (open circuit).

#### Question B

Now, suppose we short-circuit the compound battery. (This is very dangerous. NEVER do this to a large battery, such as a lead-acid battery in a car, or to a lithium-ion battery from your laptop. You MAY live to regret it, but you may not.) What is the current (in Amperes) you should expect to go through the short circuit?

#### Solution

Use I = VR for both segments, then sum

In [12]:
V1 = 1.5
V2 = 1.5
R1 = 0.25
R2 = 0.32

# Get I=V/R for each side
I1 = V1/R1
I2 = V2/R2
print('I1:',I1,'amps')
print('I2:',I2,'amps')

# Then sum
I = I1 + I2
print('I:',I,'amps')

I1: 6.0 amps
I2: 4.6875 amps
I: 10.6875 amps


#### Question C

We can think of this combination as a bigger battery of the same voltage as the two component batteries. What is the equivalent resistance (in Ohms) of the compound battery? (Hint: you have the voltage with nothing connected and the current when shorted out.)

#### Solution

Just use R=V/I, with V=1.5 and the current as the value we just computed

In [13]:
V = V1
R = V/I
print(R, 'ohms')

0.14035087719298245 ohms


#### Question D

Now, suppose that the voltages of the two component batteries are not quite the same. For example, suppose that . Then when we hook the two batteries together current will flow and the higher voltage battery will charge the lower voltage one. What is the current (in Amperes) that will flow?

#### Solution

The circuit is NOT short-circuited this time, unlike in question B (the question should have been more clear).

So, for the open circuit, we need to compute I = V/R with separate voltages. Voltages subtract, resistances add (because we are doing KVL in a loop):

In [14]:
V2 = 1.6

I = (V2-V1)/(R2+R1)
print(I,'amps')

0.1754385964912282 amps


## Part 2: Circuit Analysis Toolchest

### S2E1: Circuit Topology

In the bridge network shown below:

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/5b95533e27a256a7aa6c84f44818b055/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_bridge.gif)

#### Q&A

How many nodes? 4

How many KCL equations are independent? 3

How many loops? 7 

How many KVL equations are independent? 3

Notice that in any circuit there is always one more node than there are independent KCL equations. If two nodes share a branch, the current entering one node from that branch is the negative of the current entering the other node from that branch. So the sum of the currents entering two nodes does not count current going from one to the other. As a consequence, the sum of all the currents entering all but one of the nodes is the same as the current entering the remaining node. So the KCL equation for that node is the sum of the KCL equations for all the other nodes.

Also, in a circuit with more than one loop there are always more loops than KVL equations. The argument is analogous with the argument for nodes: if two loops share a branch, and we count the voltages counterclockwise in each loop, then the KVL equation for the branches containing both subloops is the sum of the KVL equations for each subloop. This bigger equation does not count the current in the shared branch, since it is of opposite sign in the two subloops.

### S2E2: Associated Reference Directions

The figure below shows two identical circuits connecting a 6V battery to an 18ohm resistor. The difference is that we chose to measure the voltages and currents in the two circuits differently: we used a different coordinate system of voltages and currents in our measurements.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/a30c0c774563b70bacded47e38db89aa/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_ard.gif)

#### Q&A

What is the voltage (in Volts) V1 measured across the battery? 6V

What is the voltage (in Volts) V2 measured across the resistor? 6V

What is the current (in Amperes) I1 measured entering the battery? -0.333amps

What is the current (in Amperes) I2 measured entering the resistor? 0.333amps

In [15]:
V1 = 6
V2 = 6
I1 = -0.333
I2 = 0.333
P1 = V1*I1
P2 = V2*I2
print('P1:', P1, 'watts')
print('P2:', P2, 'watts')

P1: -1.9980000000000002 watts
P2: 1.9980000000000002 watts


What is the power (in Watts) P1=V1*I1 entering the voltage source? -2W

What is the power (in Watts) P2=V2*I2 entering the resistor? 2W

**The next 4 answers are just the inverse of the first circuit**

What is the voltage (in Volts) V3 measured across the battery? -6V

What is the voltage (in Volts) V4 measured across the resistor? -6V

What is the current (in Amperes) I3 measured entering the battery? 0.333

What is the current (in Amperes) I4 measured entering the resistor? -0.333

What is the power (in Watts) P3=V3*I3 entering the voltage source? -2W

What is the power (in Watts) P4=V4*I4 entering the resistor? 2W

### S2E3: Using KVL, KCL, and VI constraints

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/18f86f86e37206d7925d132337255859/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_four.gif)

In the network shown you are given that:

In [16]:
V=2.0
I=3.0
R1=4.0
R2=5.0

# We can also get the conductances to make the math a little more straightforward:
G1 = 1/R1
G2 = 1/R2

#### Q&A

We can formulate the following two equations, so that we can solve the two unknowns, i1 and V2:

In [17]:
Math(r'i_1 = \frac{V_2}{R_2} - I')

<IPython.core.display.Math object>

In [18]:
Math(r'i_1 = \frac{V - v_2}{R_1}')

<IPython.core.display.Math object>

Solving for i1, we get the equation:

In [19]:
Math(r'i_1 = \frac{V-IR_2}{R_1+R_2}')

<IPython.core.display.Math object>

In [22]:
I1 = (V-I*R2)/(R1+R2)
print('i1:', I1, 'amps')

i1: -1.4444444444444444 amps


We just sub i1 back into one of the old equations to get v2:

In [24]:
V2 = I1*R2 + I*R2 
print('v2:', V2, 'volts')

v2: 7.777777777777778 volts


From here, it is trivial to answer the questions:

What is the voltage (in Volts) V2 across the resistor with resistance R2? 7.78 volts

What is the power (in Watts) dissipated by the resistor with resistance R2?

In [38]:
P = (I1+I)**2 * R2
print('P:', P, 'watts')

P: 12.098765432098766 watts


What is the current (in Amperes) I1 through the resistor with resistance R1? -1.44 amps

What is the power (in Watts) dissipated by the resistor with resistance R1?

In [28]:
P = I1**2 * R1
print('P:', P, 'Watts')

P: 8.345679012345679 Watts


What is the power (in Watts) supplied by the voltage source?

In [30]:
P = I1 * V
print('P:', P, 'Watts')

P: -2.888888888888889 Watts


What is the power (in Watts) supplied by the current source?

In [31]:
P = I * V2
print('P:', P, 'Watts')

P: 23.333333333333332 Watts


You should observe that the sum of the power supplied by the sources is the sum of the power dissipated by the resistors. If this is not true you have done something wrong.

### S2E4: Series and Parallel

In the network shown you are given:

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/43051f019474087b12193590fcdb2673/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_ladder.gif)

In [42]:
V = 2
R1 = 4
R2 = 4
R3 = 2
R4 = 2

#### Q&A

What is the equivalent resistance (in Ohms) of the series combination of R3 and R4?

In [44]:
R34 = R3 + R4
print('R34:', R34, 'ohms')

R34: 4 ohms


What is the equivalent resistance (in Ohms) of the parallel combination of R2 and the series combination of R3 and R4?

In [50]:
R234 = (1/R34 + 1/R2)**-1
print('R234:', R234, 'ohms')

R234: 2.0 ohms


What is the equivalent resistance (in Ohms) of the series combination of R1 and the parallel combination of R2 and the series combination of R3 and R4?

In [51]:
R1234 = R1 + R234
print('R1234:', R1234, 'ohms')

R1234: 6.0 ohms


What is the current (in Amperes) I1?

In [52]:
I1 = V/R1234
print('I1:', I1, 'amps')

I1: 0.3333333333333333 amps


What is the voltage (in Volts) across the resistor R2?

In [54]:
V2 = I1*R234
print('V2:', V2, 'volts')

V2: 0.6666666666666666 volts


What is the voltage (in Volts) across the resistor R4?

In [60]:
# R3 and R4 are two resistors in series, so the voltage is half the voltage of the resistor in parallel (R2)
V4 = V2/2
print('V4:', V4, 'volts')

V4: 0.3333333333333333 volts


### Node analysis practice, part 1

In the circuit shown, what is the node potential of the node labeled "a" relative to the ground indicated?

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/f0c9a7536a40a9b3d6619956b63b215c/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_bridge-with-is-1.gif)

#### Answer

This is just V0, because the Node is connected directly to the voltage source

### Node analysis practice, part 2

In the circuit shown, what is the current going up through the resistor with resistance R1 in terms of the node potentials?

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/743d24a788c5a2143e243dcf817e64ee/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_bridge-with-is-2.gif)

#### Answer

The current is (e1-V0)/R1

### Node analysis practice, part 3

Write the expression that represents the sum of the currents leaving the node with potential e2 in the circuit given.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/743d24a788c5a2143e243dcf817e64ee/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_bridge-with-is-2.gif)

#### Answer

(e2-V0)/R4+(e2-e1)/R3+(e2)/R5-I1

### S2E5: Node Method

In the network shown you are given:

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/0dab32afe047cf9b503b0548077df164/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_fnode.gif)

In [61]:
V = 5
I = 3
R1 = 3
R2 = 5

There are three node potentials labeled in this network e1, e2, and e3. We can choose any node to be the reference node (the ground) from which voltages are measured, setting the corresponding node potential to zero. (Only voltages, which are differences of node potentials, are physically meaningful.) Let's start by choosing the ground at the negative terminal of the voltage source, so e3=0. 

#### Q&A

**What is the value (in Volts) of the node potential e1?**

This is just V, so 5V

In [70]:
e3 = 0
e1 = V
print('e1:',e1,'volts')

e1: 5 volts


**Now there is one remaining unknown node, with potential e2. Write a node equation for that node. What is the value (in Volts) of e2?**

Noting that e3=0, KCL is: 

In [62]:
Math(r'0 = I - e2/R2 - (e2-e1)/R1')

<IPython.core.display.Math object>

Rearrange to get:

In [63]:
Math(r'e_2 = \frac{I + \frac{e_1}{R_1}}{\frac{1}{R_2} + \frac{1}{R_1}}')

<IPython.core.display.Math object>

e1 is equal to V, so:

In [67]:
e2 = (I+e1/R1)/(1/R2+1/R1)
print('e2:',e2,'volts')

e2: 8.75 volts


**What is the voltage (in Volts) V2 across the resistor with resistance R2?**

In [72]:
V2 = e2 -e3
print('V2:', V2, 'volts')

V2: 8.75 volts


**Now, let's start again. Suppose we chose a different node as the ground reference. Let's choose the top terminal of the current source, where it connects to the two resistors. So now e2 = 0, and there are two remaining nodes. But if we know e3 then we know that e1 = e3 + V, so we still have only one unknown. What is the value (in Volts) of the node potential e3?**

Since we just flipped it vertically, e3 is just the negative of e2 above: -8.75 V

**Now, what is the voltage (in volts) V2 across the resistor with resistance R2?**

8.75 V

### S2E6: Modeling

Joe has a barn that is 113.0 feet from his house. He needs to supply 1000 Watts at 240V to a resistive load at his barn from the 60Hz power line at his house. Note that the circuit from the house to the barn requires two lengths of the interconnecting wire. He proposes to use number 12 AWG wire to connect his house to his barn. (AWG is American Wire Gauge: a specification of the size of the wire. For more information see.) Number 12 AWG copper wire has a resistance of 1.588ohms per 1000 feet.

![](https://prod-edxapp.edx-cdn.org/assets/courseware/v1/aceb78ef8e7d5b4322afef5ac7244cb9/asset-v1:MITx+6.002.1x_1+2T2016+type@asset+block/images_circuits_house-barn.gif)

#### Q&A

**What is the total resistance (in Ohms) of the transmission line?**

In [74]:
num = 2
dist = 113.0/1000
R = num * dist * 1.588
print('R:', R, 'ohms')

R: 0.35888800000000004 ohms


**What is the resistance (in Ohms) of Joe's load at his barn?**

In [77]:
V = 240
P = 1000
Rload = V**2/P
print('Rload:', Rload, 'ohms')

Rload: 57.6 ohms


**What is the voltage drop (in Volts) from the house to the load at the barn and back to the house due to the resistance in Joe's transmission line, assuming that Joe is able to get 240V across the load at the barn?**

In [81]:
Iload = V/Rload
print('Iload:', Iload, 'amps')
Vline = Iload * R
print('Vline:', Vline, 'volts')

Iload: 4.166666666666667 amps
Vline: 1.495366666666667 volts
